### PR TITLE
[Doc] Analyze external table

### PR DESCRIPTION
**Pull request by evelynzhaojie**
Why I'm doing:

What I'm doing:
support collecting statistics of external tables in v3.2
add an FE config of CBO added in v3.2
add a session variable that was not written in doc

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/using_starrocks/Cost_based_optimizer.md [hunk 1]
```diff
@@ -399,40 +430,216 @@ DROP STATS tbl_name
...
-| histogram_sample_ratio               | FLOAT    | 0.1               | The sampling ratio for a histogram.                          |
-| histogram_max_sample_row_count       | LONG     | 10000000          | The maximum number of rows to collect for a histogram.       |
 | statistic_manager_sleep_time_sec     | LONG     | 60                | The interval at which metadata is scheduled. Unit: seconds. The system performs the following operations based on this interval:Create tables for storing statistics.Delete statistics that have been deleted.Delete expired statistics. |
 | statistic_analyze_status_keep_second | LONG     | 259200            | The duration to retain the history of collection tasks. The default value is 3 days. Unit: seconds. |
 
+## Session variable
+
+`statistic_collect_parallel`: Used to adjust the parallelism of statistics collection tasks that can run on BEs. Default value: 1. You can increase this value to speed up collection tasks.
+
+## Collect statistics for external tables
```
Comment (EsoragotoSpirit) [inline docs/en/using_starrocks/Cost_based_optimizer.md:458]: 以后我们对于 external catalog 里的表都叫做外表（external table）了吗？会不会和外表的概念冲突？这块我们要统一一下吗？

File: docs/en/using_starrocks/Cost_based_optimizer.md [hunk 2]
```diff
@@ -399,40 +430,216 @@ DROP STATS tbl_name
...
+   update_time: 2023-12-01 14:57:27.137000
+```
+
+### Limits
+
+The following limits apply when you collect statistics from external tables:
+
+1. Only full collection is supported. Sampled collection and histogram collection are not supported.
+2. You can collect statistics of only Hive, Iceberg, and Hudi tables.
+3. For automatic collection tasks, you can only collect statistics of a specific table. You cannot collect statistics of all tables in a database or statistics of all databases in a catalog.
```
Comment (EsoragotoSpirit) [inline docs/en/using_starrocks/Cost_based_optimizer.md:487]: in a catalog  ---> in an external catalog?

File: docs/en/using_starrocks/Cost_based_optimizer.md [hunk 3]
```diff
@@ -399,40 +430,216 @@ DROP STATS tbl_name
...
+4. For automatic collection tasks, StarRocks can detect whether data in Hive and Iceberg tables are updated and if so, collect statistics of only partitions whose data is updated. StarRocks  cannot perceive whether data in Hudi tables are updated and can only perform periodic full collection.
```
Comment (EsoragotoSpirit) [inline docs/en/using_starrocks/Cost_based_optimizer.md:488]: extra space

File: docs/en/using_starrocks/Cost_based_optimizer.md [hunk 4]
```diff
@@ -399,40 +430,216 @@ DROP STATS tbl_name
...
++----------------------+-------------+---------+------+---------------------+------------+---------+
+| Database             | Table       | Columns | Type | UpdateTime          | Properties | Healthy |
++----------------------+-------------+---------+------+---------------------+------------+---------+
+| hive_catalog.tn_test | ex_hive_tbl | k1      | FULL | 2023-12-04 16:37:46 | {}         |         |
++----------------------+-------------+---------+------+---------------------+------------+---------+
+```
+
+#### Cancel a collection task
+
+Cancels a Running collection task.
```
Comment (EsoragotoSpirit) [inline docs/en/using_starrocks/Cost_based_optimizer.md:557]: running

File: docs/zh/using_starrocks/Cost_based_optimizer.md [hunk 5]
```diff
@@ -425,36 +457,206 @@ ANALYZE TABLE tbl_name DROP HISTOGRAM ON col_name [, col_name];
...
+
+```sql
+KILL ANALYZE <ID>
+```
+
+任务 ID 可以在 SHOW ANALYZE STATUS 中查看。
+
+### 自动收集
+
+创建一个自动收集任务，StarRocks 会周期性检查收集任务是否需要执行，默认检查时间为 5min。Hive 和 Iceberg 发现有数据更新时，才会自动执行一次采集任务。Hudi 目前不支持感知数据更新，所以只能周期性采集（采集周期由收集线程的时间间隔和用户设置的采集间隔决定，参考下面的参数调整）。
```
Comment (EsoragotoSpirit) [inline docs/zh/using_starrocks/Cost_based_optimizer.md:586]: 参数(Property)

File: docs/en/using_starrocks/Cost_based_optimizer.md [hunk 6]
```diff
@@ -399,40 +430,216 @@ DROP STATS tbl_name
...
+
+  Threshold to determine whether an external table is a small table during automatic collection. Default: 10000000.
+
+- statistic_auto_collect_small_table_interval
+
+  The interval for collecting statistics of small tables. Unit: seconds. Default: 0.
+
+- statistic_auto_collect_large_table_interval
+
+  The interval for collecting statistics of large tables. Unit: seconds. Default: 12 h.
```
Comment (EsoragotoSpirit) [inline docs/en/using_starrocks/Cost_based_optimizer.md:585]: Default: 43200 (12 hours).

File: docs/zh/using_starrocks/Cost_based_optimizer.md [hunk 7]
```diff
@@ -425,36 +457,206 @@ ANALYZE TABLE tbl_name DROP HISTOGRAM ON col_name [, col_name];
...
+
+  自动收集中，用来判断一个外表是否为小表的行数门限，默认值为 10000000 行。该参数 3.2 版本引入。
+
+- statistic_auto_collect_small_table_interval
+
+  自动收集中小表的收集间隔，默认值为 0，单位：秒。
+
+- statistic_auto_collect_large_table_interval
+
+  自动收集中大表的收集间隔，默认值为 12h，单位：秒。
```
Comment (EsoragotoSpirit) [inline docs/zh/using_starrocks/Cost_based_optimizer.md:602]: 默认值：43200 (12 hours)

File: docs/en/using_starrocks/Cost_based_optimizer.md (hunk 1 above)
Comment (evelynzhaojie) [reply docs/en/using_starrocks/Cost_based_optimizer.md]: 统一修改了外表这个用词

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by hellolilyliuyi — APPROVED** [review]

**Pull request merged**